src/main/java/org/apache/maven/plugins/assembly/archive/DefaultAssemblyArchiver.java

```diff
@@ -351,6 +349,23 @@ protected Archiver createArchiver(
         return archiver;
     }
 
+    private void configureJarArchiver(
+            JarArchiver archiver, String mergeManifestMode, MavenArchiveConfiguration configuration) {
+
+        if (mergeManifestMode != null) {
+            archiver.setFilesetmanifest(JarArchiver.FilesetManifestConfig.valueOf(mergeManifestMode));
+        }
+
+        archiver.setMinimalDefaultManifest(true);
+
+        if (configuration != null) {
+            archiver.setCompress(configuration.isCompress());
+            archiver.setForced(configuration.isForced());
+            archiver.setIndex(configuration.isIndex());
```

Comment:
I am confused, where do these come from?

Reply:
It comes from
https://github.com/apache/maven-archiver/blob/master/src/main/java/org/apache/maven/archiver/MavenArchiver.java#L576-L593

Reply:
But these aren't related to this change, no? I am still searching for the connection between the issue title and this hunk.

Reply:
Ok, I did it by the way - can be in separate issue 😄

Reply:
does this have to be checked against mergeManifestmode or not?

Reply:
As I wrote code was copied from maven-archiver, for this change I will remove it